front/pages/api/v1/w/[wId]/vaults/[vId]/data_source_views/[dsvId]/index.ts

```diff
@@ -0,0 +1,180 @@
+ *         in: path
+ *         required: true
+ *         schema:
+ *           type: string
+ *       - name: vId
+ *         in: path
+ *         required: true
+ *         schema:
+ *           type: string
+ *       - name: dsvId
+ *         in: path
+ *         required: true
+ *         schema:
+ *           type: string
+ *     requestBody:
+ *       required: true
+ *       content:
+ *         application/json:
+ *           schema:
+ *             $ref: '#/components/schemas/PatchDataSourceViewSchema'
+ *     responses:
+ *       '200':
+ *         description: Successful response
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 dataSourceView:
+ *                   $ref: '#/components/schemas/DataSourceViewType'
+ *       '400':
+ *         description: Invalid request body
+ *       '403':
+ *         description: Unauthorized
+ *       '404':
+ *         description: Data source view not found
+ *       '405':
+ *         description: Method not allowed
+ *       '500':
+ *         description: Internal server error
+ */
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<
+    WithAPIErrorResponse<
+      GetOrPostDataSourceViewsResponseBody | PatchDataSourceViewResponseBody
+    >
+  >,
+  auth: Authenticator
+): Promise<void> {
+  const dataSourceView = await DataSourceViewResource.fetchById(
+    auth,
+    req.query.dsvId as string
+  );
+
+  switch (req.method) {
+    case "GET":
+      return res.status(200).json({
+        dataSourceView: dataSourceView.toJSON(),
+      });
+
+    case "PATCH":
+      const patchBodyValidation = PatchDataSourceViewSchema.decode(req.body);
+      if (isLeft(patchBodyValidation)) {
+        const pathError = reporter.formatValidationErrors(
+          patchBodyValidation.left
+        );
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message: `invalid request body: ${pathError}`,
+          },
+        });
+      }
+
+      const { right: body } = patchBodyValidation;
+
+      let updateResultRes: Result<undefined, Error>;
+      if ("parentsIn" in body) {
+        const { parentsIn } = body;
+        updateResultRes = await dataSourceView.setParents(parentsIn);
+      } else {
+        const { parentsToAdd, parentsToRemove } = body;
+        updateResultRes = await dataSourceView.updateParents(
+          parentsToAdd ?? [],
+          parentsToRemove ?? []
+        );
+      }
+
+      if (updateResultRes.isErr()) {
+        return apiError(req, res, {
+          status_code: 500,
+          api_error: {
+            type: "internal_server_error",
+            message: "The data source view cannot be updated.",
+          },
+        });
+      }
+
+      return res.status(200).json({
+        dataSourceView: dataSourceView.toJSON(),
+      });
```

Comment:
can we share the code with the private handler by extracting a method (including permission checks ?)